Stock.Api/Controllers/ProductController.cs

```diff
@@ -0,0 +1,137 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using Stock.Api.DTOs;
+using Stock.Api.Extensions;
+using Stock.AppService.Services;
+using Stock.Model.Entities;
+
+namespace Stock.Api.Controllers
+{
+    [Produces("application/json")]
+    [Route("api/product")]
+    [ApiController]
+    public class ProductController : ControllerBase
+    {
+        private readonly ProductService service;
+        private readonly IMapper mapper;
+        
+        public ProductController(ProductService service, IMapper mapper)
+        {
+            this.service = service;
+            this.mapper = mapper;
+        }
+
+        /// <summary>
+        /// Permite recuperar todas las instancias
+        /// </summary>
+        /// <returns>Una colección de instancias</returns>
+        [HttpGet]
+        public ActionResult<IEnumerable<ProductDTO>> Get()
+        {
+            return this.mapper.Map<IEnumerable<ProductDTO>>(this.service.GetAll()).ToList();
+        }
+
+        /// <summary>
+        /// Permite recuperar una instancia mediante un identificador
+        /// </summary>
+        /// <param name="id">Identificador de la instancia a recuperar</param>
+        /// <returns>Una instancia</returns>
+        [HttpGet("{id}")]
+        public ActionResult<ProductDTO> Get(string id)
+        {
+            return this.mapper.Map<ProductDTO>(this.service.Get(id));
+        }
+
+        /// <summary>
+        /// Permite crear una nueva instancia
+        /// </summary>
+        /// <param name="value">Una instancia</param>
+        [HttpPost]
+        public ActionResult Post([FromBody] ProductDTO value)
+        {
+            TryValidateModel(value);
+            
+            try
+            {
+                var product = this.mapper.Map<Product>(value);
+                this.service.Create(product);
```

Comment:
product.ProductType = this.productTypeService.Get(value.ProductTypeId.ToString());
Te estaría faltando esa linea...

Reply:
@JonathanLoscalzo gracias por los comentarios!! :)
Te quería consultar como haces el binding entre el Product y el ProductType, estuve viendo las clases, tanto del modelo como del DTO, y ya estan creadas las variables que vinculan Product y ProductType.

**_Tenés que asignarle a la entidad Product el ProductType, buscarlo en la db con el servicio y asociarlo.
Esa tarea se puede hacer en un servicio en lugar del controller, es decir delegar._**  <-- ¿Esto lo haces en el ProductService en el método Get? O sea habría que crear una instancia del ProductTypeService en el ProductService y levantar la categoría según el ID del ProductType en el Product?  Te paso una idea de lo que se me ocurre, seguramente no sea correcto:

        public Product Get(string id)
        {
            Product aux = this.Repository.GetById(id); <-- Levanto el producto del repo.
            ProductTypeService ptService = new ProductTypeService(); <-- Creo el service de ProductType para levantar el ProductType del Product (Esto me hace ruido!)
            ProductType ptAux = ptService.Get(aux.ProductType.Id); <-- Lo levanto del repo.
            aux.ProductType = ptAux; <-- Vinculo el Product con su ProductType y lo devuelvo.
            return aux;
        }

Disculpa, estoy un poco perdido con como vincularlos!! Muchas gracias! :)

Reply:
> ¿Esto lo haces en el ProductService en el método Get? O sea habría que crear una instancia del ProductTypeService en el ProductService y levantar la categoría según el ID del ProductType en el Product?

Podés hacerlo donde quieras. Digamos, yo te dije que lo hagas en el crear, pero podés hacerlo en el GetAll también.

Reply:
Tené en cuenta también que en caso de modificar la categoría, se debería ver modificada en el listado de productos...

Podés tomar varios enfoques ahí...
- no tener en cuenta la descripción y siempre recuperar la última versión de categoria/provider (con tu get(id))
- podés hacer que cada vez que insertas una categoría, modifique todas las categorías a productos asociados.
- podés usar dbref. 

Con respecto al 2do punto... como está el modelo ahora se está guardando así: 
Documento "ProductTypes", Shape: { _id, initials, description }
Documento "Products", Shape: { _id, name, costPrice, salePrice, Stock, ProductType: { _id, initials, description } } 

Se está duplicando la categoría... entonces hay que hacer un join en algún lado. Vos ahí me pasaste un posible join para recuperar siempre la última descripción de ProductTypes. 

No se si fui muy claro, pero quiero destacar que Product tiene anidado (nested) ProductType y ese no es la última versión.